uaclient/tests/test_system.py

```diff
@@ -155,23 +193,139 @@ class TestGetKernelInfo:
+            (
+                os.uname_result(
+                    [
+                        "",
+                        "",
+                        "",
+                        "#20-Ubuntu SMP PREEMPT_DYNAMIC Thu Apr  6 07:48:48 UTC 2023",  # noqa: E501
+                        "",
+                    ]
+                ),
+                mock.sentinel.changelog_timestamp,
+                datetime.datetime(
+                    2023, 4, 6, 7, 48, 48, tzinfo=datetime.timezone.utc
+                ),
+            ),
+            (
+                os.uname_result(
+                    [
+                        "",
+                        "",
+                        "",
+                        "#33~22.04.1-Ubuntu SMP PREEMPT_DYNAMIC Mon Jan 30 17:03:34 UTC 2",  # noqa: E501
+                        "",
+                    ]
+                ),
+                mock.sentinel.changelog_timestamp,
+                mock.sentinel.changelog_timestamp,
+            ),
+            (
+                os.uname_result(["", "", "", "corrupted", ""]),
+                mock.sentinel.changelog_timestamp,
+                mock.sentinel.changelog_timestamp,
+            ),
+        ],
+    )
+    @mock.patch("uaclient.system._get_kernel_changelog_timestamp")
+    def test_get_kernel_build_date(
+        self,
+        m_get_kernel_changelog_timestamp,
+        uname_result,
+        changelog_timestamp,
+        expected,
+    ):
+        m_get_kernel_changelog_timestamp.return_value = changelog_timestamp
+        assert expected == system._get_kernel_build_date(uname_result)
+
+    @pytest.mark.parametrize(
+        [
+            "uname_result",
+            "is_container",
+            "stat_result",
+            "expected_stat_call_args",
+            "expected",
+        ],
+        [
+            (
+                None,
+                True,
+                None,
+                [],
+                None,
+            ),
+            (
+                os.uname_result(["", "", "version-here", "", ""]),
+                False,
+                Exception(),
+                [
+                    mock.call(
+                        "/usr/share/doc/linux-image-version-here/changelog.Debian.gz"  # noqa: E501
+                    )
+                ],
+                None,
+            ),
+            (
+                os.uname_result(["", "", "version-here", "", ""]),
+                False,
+                [os.stat_result([0, 0, 0, 0, 0, 0, 0, 0, 1680762951, 0])],
+                [
+                    mock.call(
+                        "/usr/share/doc/linux-image-version-here/changelog.Debian.gz"  # noqa: E501
+                    )
+                ],
+                datetime.datetime(
+                    2023, 4, 6, 6, 35, 51, tzinfo=datetime.timezone.utc
+                ),
+            ),
+        ],
+    )
+    @mock.patch("os.stat")
+    @mock.patch("uaclient.system.is_container")
+    def test_get_kernel_changelog_timestamp(
```

Comment:
Should we add a test that confirm we are failling back to `_get_kernel_changelog_timestamp` if we fail to get the date from the uname result

Reply:
I'm testing this using mock sentinels in the test above this one. See lines: 239-256

Reply:
Indeed, I missed that strategy